virtualization-formulas/sriov/states/init.sls

```diff
@@ -0,0 +1,33 @@
+{% if pillar["devices"] is mapping %}
+sriov_unit_file:
+  file.managed:
+    - name: /etc/systemd/system/sriov.service
+    - contents: |
+        [Unit]
+        Description = Activate SR-IOV virtual interfaces
+
+        [Service]
+        Type=oneshot
+        ExecStart=\
+  {%- for eth, count in pillar["devices"].items() %}
+          sh -c "echo {{ count }} >/sys/class/net/{{ eth }}/device/sriov_numvfs" ; \
+  {%- else %}
+          sh -c 'for file in `ls /sys/class/net/*/device/sriov_numvfs`; do echo 0 >$file; done'
+  {%- endfor %}
+
+        [Install]
+        WantedBy = network-online.target
```

Comment:
I think that after adding service, you will need to call `systemctl daemon-reload` to be picked up by systemd.

Reply:
oh right, I'll add it